client/mysqlbinlog.cc

```diff
@@ -2111,6 +2129,24 @@ static int parse_args(int *argc, char*** argv)
 }
 
 
+/**
+  Print time into progress log.
+
+  @param mytime timesnapt to be print
+*/
+static void print_progress_log_time(my_time_t mytime)
+{
+  struct tm *res = localtime((time_t *)&mytime);
+  fprintf(progress_file, "Time: %04d%02d%02d %2d:%02d:%02d",
```

Comment:
`localtime` is likely to be ambiguous to the _consumer_ of the log, who may be in a different timezone, and standard [ISO8601 format](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations) should be more universally parseable.

So, how about this, to use UTC and ISO8601 format… ?

```c
struct tm *res = gmtime((time_t *)&mytime);
fprintf(progress_file, "Time: %04d%02d%02dT%2d:%02d:%02dZ",
```

Reply:
That's a great suggestion! Thanks! updated in the commit.

Reply:
@dlenski, that does not look like ISO 8601 format, because it is missing the dashes between years and months.
`"%04d-%02d-%02dT%02d:%02d:%02dZ"` (note: also a leading zero for the hours) would be better. And yes, you made a very good point about the Z and `gmtime()`.

Reply:
@dr-m Nice catch! Not sure how I missed it.. Updated in commit to use `"%04d-%02d-%02dT%02d:%02d:%02dZ"`